### PR TITLE
Allow std::move to work with Core classes

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

These instances were caught by a clang static analysis.

#### PR validation:

The code compiles.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

File: FWCore/Concurrency/interface/WaitingTaskHolder.h [hunk 1]
```diff
@@ -48,6 +48,12 @@ namespace edm {
       return *this;
     }
 
+    WaitingTaskHolder& operator=(WaitingTaskHolder&& iRHS) {
+      WaitingTaskHolder tmp(std::move(iRHS));
```
Comment (fwyzard) [inline FWCore/Concurrency/interface/WaitingTaskHolder.h:52]: std::move is unnecessary, as iRHS is already a && ?

Comment (Dr15Jones) [reply FWCore/Concurrency/interface/WaitingTaskHolder.h:52]: I don't believe that is guaranteed

Comment (Dr15Jones) [reply FWCore/Concurrency/interface/WaitingTaskHolder.h:52]: I did a quick test with compiler explorer using -O0 and the compiler used the copy constructor if I did not do an explicit `std::move`

Comment (makortel) [reply FWCore/Concurrency/interface/WaitingTaskHolder.h:52]: My understanding (could be wrong) is that the `&&` in `iRHS` is about to what `iRHS` can bind to, but the `iRHS` itself is an lvalue (or something similar, but not rvalue).

Comment (Dr15Jones) [reply FWCore/Concurrency/interface/WaitingTaskHolder.h:52]: I tested using -O2 and -O3 and also got the same result.

**Comment by makortel** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**